caffe2-cnn/mnist_nn_cpu.py

```diff
@@ -0,0 +1,181 @@
+import csv
+import math
+import numpy as np
+import os
+import shutil
+import subprocess
+import urllib2
+from caffe2.python import workspace, core, cnn, net_drawer, visualize
+from sigopt import Connection
+
+current_folder = os.path.join(os.getcwd(), 'caffe2_mnist')
+
+data_folder = os.path.join(current_folder, 'tutorial_data', 'mnist')
+root_folder = os.path.join(current_folder, 'tutorial_files', 'tutorial_mnist')
+image_file_train = os.path.join(data_folder, "train-images-idx3-ubyte")
+label_file_train = os.path.join(data_folder, "train-labels-idx1-ubyte")
+image_file_test = os.path.join(data_folder, "t10k-images-idx3-ubyte")
+label_file_test = os.path.join(data_folder, "t10k-labels-idx1-ubyte")
+
+conn = Connection(client_token='SIGOPT_API_TOKEN')
+
+# Get the dataset if it is missing
+def download_dataset(url, path):
+  import requests, zipfile, StringIO
+  r = requests.get(url, stream=True)
+  z = zipfile.ZipFile(StringIO.StringIO(r.content))
+  z.extractall(path)
+
+# Load training/testing data into db
+def generate_db(image, label, name):
+  name = os.path.join(data_folder, name)
+  if not os.path.exists(name):
+    syscall = "/usr/local/binaries/make_mnist_db --channel_first --db leveldb --image_file " + image + " --label_file " + label + " --output_file " + name
+    subprocess.check_call(syscall, shell=True)
+
+def data_setup():
+  if not os.path.exists(data_folder):
+    os.makedirs(data_folder)
+  if not os.path.exists(label_file_train):
+    download_dataset("https://s3.amazonaws.com/caffe2/datasets/mnist/mnist.zip", data_folder)
+
+  if os.path.exists(root_folder):
+    shutil.rmtree(root_folder)
+
+  os.makedirs(root_folder)
+  workspace.ResetWorkspace(root_folder)
+
+  # (Re)generate the leveldb database (known to get corrupted...)
+  generate_db(image_file_train, label_file_train, "mnist-train-nchw-leveldb")
+  generate_db(image_file_test, label_file_test, "mnist-test-nchw-leveldb")
+
+# Define the SigOpt experiment
+def setup_sigopt_experiment(conn):
```

Comment:
Since we've defined conn as a global we don't need to pass it into this function